File: hosung/leetcode/20. Valid Parentheses.py
class Solution(object):
    def isValid(self, s):
        a = {')':'(', ']':'[', '}':'{'}
        st = []
        for i in s:
            if i in '([{':
                st.append(i)
            else:
                if st and a[i] == st[-1] :
                    st.pop()
                else:
                    return False
        return len(st) ==0

File: taejoong/leetcode/20_ValidParentheses.java
class Solution {
    public boolean isValid(String s) {

        Stack<Character> stack = new Stack<>();
        char[] arr = s.toCharArray();

        for(char c : arr) {

            if(c == '(') {
                stack.push(')');
	    } else if(c == '{') {
                stack.push('}');
            } else if(c == '[') {
                stack.push(']');
            } else if(stack.isEmpty() || c != stack.pop()) {
                return false;
            }
        }
        return stack.isEmpty();
    }
}